packages/repository/src/relations/has-many/has-many-through.helpers.ts

```diff
@@ -68,21 +67,70 @@ export function createTargetConstraint<
   relationMeta: HasManyThroughResolvedDefinition,
   throughInstances: Through | Through[],
 ): DataObject<Target> {
+  const fkValues = createFkValues(relationMeta, throughInstances);
   const targetPrimaryKey = relationMeta.keyTo;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const constraint: any = {
+    [targetPrimaryKey]: Array.isArray(fkValues) ? {inq: fkValues} : fkValues,
+  };
+  return constraint;
+}
+
+/**
+ * Returns a target fk or an array of target fks of the given throughInstances.
+ *
+ * @param relationMeta
+ * @param throughInstances
+ *
+ * @example
+ * ```ts
+ * const resolvedMetadata = {
+ *  // .. other props
+ *  keyFrom: 'id',
+ *  keyTo: 'id',
+ *  through: {
+ *    model: () => CategoryProductLink,
+ *    keyFrom: 'categoryId',
+ *    keyTo: 'productId',
+ *  },
+ * };
+ * createFkValues(resolvedMetadata,{
+        id: 2,
+        categoryId: 2,
+        productId: 8,
+      });
+ * >>> 8
+* createFkValues(resolvedMetadata, [
+      {
+        id: 2,
+        categoryId: 2,
+        productId: 8,
+      }, {
+        id: 1,
+        categoryId: 2,
+        productId: 9,
+      }
+  ]);
+  >>> [8, 9]
+ */
+export function createFkValues<Through extends Entity, TargetID>(
+  relationMeta: HasManyThroughResolvedDefinition,
+  throughInstances: Through | Through[],
```

Comment:
Is it necessary to support both `Trough` and `Through[]` variants? I noticed this design is affecting multiple helpers. Wouldn't it be simpler to support a single value only (`Through`) and let the caller of this function to deal with the array case by calling `instances.map(i => createFkValues(relationMeta, i))`?

Reply:
I think `Through[]` is still needed cause `createTargetConstraint` calls it as well, and if I remove it, I need to do a lot type conversions to have the complier happy D: but I think the returned type can be simplified to `TargetID`.